File: frontend/src/lib/api/grazing.ts
import { api, buildQuery } from './client';

export interface GrazingData {
	id: number;
	date: string;
	animal_count: number | null;
	pasture_time: number | null;
	lactation_period: string | null;
}

export interface GrazingFilter {
	from_date?: string;
	till_date?: string;
	page?: number;
	per_page?: number;
}

export function listGrazing(filter: GrazingFilter = {}, signal?: AbortSignal) {
	return api<{ data: GrazingData[]; total: number; page: number; per_page: number }>(
		`/grazing${buildQuery(filter)}`,
		{ signal },
	);
}

File: backend/src/models/grazing.rs
use chrono::NaiveDate;
use serde::{Deserialize, Serialize};
use sqlx::FromRow;

#[derive(Debug, Clone, Serialize, Deserialize, FromRow, utoipa::ToSchema)]
pub struct GrazingData {
    pub id: i32,
    pub date: NaiveDate,
    pub animal_count: Option<i32>,
    pub pasture_time: Option<i32>,
    pub lactation_period: Option<String>,
}

#[derive(Debug, Deserialize, utoipa::ToSchema, utoipa::IntoParams)]
pub struct GrazingFilter {
    pub from_date: Option<NaiveDate>,
    pub till_date: Option<NaiveDate>,
    pub page: Option<i64>,
    pub per_page: Option<i64>,
}
